File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean check(int i, String s){
        if(i >= s.length()/2){
            return true;
        }

        if(s.charAt(i) != s.charAt(s.length() - i - 1)) return false;

        return check(i+1, s);
    }

    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^a-zA-Z0-9]", "");
        s = s.toLowerCase();
        return check(0, s);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        n = len(s)
        i = 0
        j = len(s) - 1

        while i<=j:
            while i<n and not s[i].isalnum():
                i+=1

            while j>=0 and not s[j].isalnum():
                j-=1

            
            if i<n and j>=0 and s[i].lower()!=s[j].lower():
                return False
            
            i+=1
            j-=1

        return True